distributed/worker.py

```diff
@@ -3201,85 +3266,6 @@ def __repr__(self):
     return msg
 
 
-def convert_args_to_str(args, max_len: int | None = None) -> str:
-    """Convert args to a string, allowing for some arguments to raise
-    exceptions during conversion and ignoring them.
-    """
-    length = 0
-    strs = ["" for i in range(len(args))]
-    for i, arg in enumerate(args):
-        try:
-            sarg = repr(arg)
-        except Exception:
-            sarg = "< could not convert arg to str >"
-        strs[i] = sarg
-        length += len(sarg) + 2
-        if max_len is not None and length > max_len:
-            return "({}".format(", ".join(strs[: i + 1]))[:max_len]
-    else:
-        return "({})".format(", ".join(strs))
-
-
-def convert_kwargs_to_str(kwargs: dict, max_len: int | None = None) -> str:
-    """Convert kwargs to a string, allowing for some arguments to raise
-    exceptions during conversion and ignoring them.
-    """
-    length = 0
-    strs = ["" for i in range(len(kwargs))]
-    for i, (argname, arg) in enumerate(kwargs.items()):
-        try:
-            sarg = repr(arg)
-        except Exception:
-            sarg = "< could not convert arg to str >"
-        skwarg = repr(argname) + ": " + sarg
-        strs[i] = skwarg
-        length += len(skwarg) + 2
-        if max_len is not None and length > max_len:
-            return "{{{}".format(", ".join(strs[: i + 1]))[:max_len]
-    else:
-        return "{{{}}}".format(", ".join(strs))
-
-
-async def run(server, comm, function, args=(), kwargs=None, wait=True):
```

Comment:
I see you are copying this code and duplicate it such that we have three different versions of this now. Is this just because of the ctx var?

If so, I could see us changing this ctx var to something that just returns the current server, s.t. there is a ctx var where one can always access the Nanny, Scheduler, Worker wherever it is called. I believe this would be useful in all sorts of cases (e.g. recently I played again with graph serialization and it would've been nice to have a way to know if one is on the scheduler or not)

I wonder if couldn't set this ctx var in `Server.{handle_stream|handle_comm|start}` and get all of this effectively for free.

Reply:
for a few reasons actually: 
* I wanted to fix up incorrectly injecting "dask_scheduler" into client.run calls and "dask_worker" into client.run_on_scheduler calls
* run calls always logged to "distributed.worker" which means it didn't show up on the scheduler logs 
* and yeah the contextvar code now makes worker.run quite a bit different from nanny.run and scheduler.run

so I think for this PR it's worth the duplication for now and we can think about cleaning up the duplication and or making a Server cvar in a future PR